moshi/moshi/run_tts.py

```diff
@@ -0,0 +1,210 @@
+import sphn
+import torch
+
+from .models.tts import TTSModel, DEFAULT_DSM_TTS_REPO, DEFAULT_DSM_TTS_VOICE_REPO
+from .models.loaders import CheckpointInfo, hf_get
+
+
+@dataclass
+class TTSRequest:
+    """Format for a single TTS request in the provided JSONL file.
+
+    Args:
+        turns: list of strings, one per turn, starting with the MAIN speaker.
+            If you want to generate single turn, just put a single entry.
+        voices: list of voice names, starting with MAIN. Put a single voice for single
+            speaker generation. A corresponding file should exist in the voice repository.
+        id: id that will be used for the output file. A wav file will be created, along
+            with a `.safetensors` and `.json` file containing debug information.
+        prefix: path to an audio file to use as prefix.
+    """
+    turns: list[str]
+    voices: list[str]
+    id: str
+
+
+def main():
+    parser = argparse.ArgumentParser(prog='moshi-tts', description='Run Moshi')
+    parser.add_argument("--out-folder", type=Path, help="Output folder for TTSed files.",
+                        default=Path('tts-outputs'))
+
+    parser.add_argument("--hf-repo", type=str, default=DEFAULT_DSM_TTS_REPO,
+                        help="HF repo in which to look for the pretrained models.")
+    parser.add_argument("--voice-repo", default=DEFAULT_DSM_TTS_VOICE_REPO,
+                        help="HF repo in which to look for pre-computed voice embeddings.")
+
+    # The following flags are only to use a local checkpoint.
+    parser.add_argument("--config", "--lm-config", dest="config", type=str, help="The config as a json file.")
+    parser.add_argument("--tokenizer", type=str, help="Path to a local tokenizer file.")
+    parser.add_argument("--mimi-weight", type=str, help="Path to a local checkpoint file for Mimi.")
+    parser.add_argument("--moshi-weight", type=str, help="Path to a local checkpoint file for Moshi.")
+
+    # The following flags will customize generation.
+    parser.add_argument("--batch-size", type=int, default=32, help="Batch size to be used for inference.")
+    parser.add_argument("--nq", type=int, default=32, help="Number of codebooks to generate.")
+    parser.add_argument("--temp", type=float, default=0.6, help="Temperature for text and audio.")
+    parser.add_argument("--cfg-coef", type=float, default=2., help="CFG coefficient.")
+
+    parser.add_argument("--max-padding", type=int, default=8, help="Max padding in a row, in steps.")
+    parser.add_argument("--initial-padding", type=int, default=2, help="Initial padding, in steps.")
+    parser.add_argument("--final-padding", type=int, default=4, help="Amount of padding after the last word, in steps.")
+    parser.add_argument("--padding-bonus", type=float, default=0.,
+                        help="Bonus for the padding logits, should be between -2 and 2, "
+                             "will change the speed of speech.")
```

Comment:
Mention the direction of the impact? E.g. "a larger padding bonus causes the speech to be slower".